Mathlib/NumberTheory/LSeries/RiemannZeta.lean

```diff
@@ -0,0 +1,246 @@
+
+lemma HurwitzZeta.cosZeta_zero : cosZeta 0 = riemannZeta := by
+  simp_rw [cosZeta, riemannZeta, hurwitzZetaEven, if_true, completedHurwitzZetaEven_zero,
+    completedCosZeta_zero]
+
+lemma HurwitzZeta.hurwitzZeta_zero : hurwitzZeta 0 = riemannZeta := by
+  ext1 s
+  simpa [hurwitzZeta, hurwitzZetaEven_zero] using hurwitzZetaOdd_neg 0 s
+
+lemma HurwitzZeta.expZeta_zero : expZeta 0 = riemannZeta := by
+  ext1 s
+  rw [expZeta, cosZeta_zero, add_right_eq_self, mul_eq_zero, eq_false_intro I_ne_zero, false_or,
+    ← eq_neg_self_iff, ← sinZeta_neg, neg_zero]
+
+/-- The Riemann zeta function is differentiable away from `s = 1`. -/
+theorem differentiableAt_riemannZeta {s : ℂ} (hs' : s ≠ 1) : DifferentiableAt ℂ riemannZeta s :=
+  differentiableAt_hurwitzZetaEven _ hs'
+#align differentiable_at_riemann_zeta differentiableAt_riemannZeta
+
+/-- We have `ζ(0) = -1 / 2`. -/
+theorem riemannZeta_zero : riemannZeta 0 = -1 / 2 := by
+  simp_rw [riemannZeta, hurwitzZetaEven, Function.update_same, if_true]
+#align riemann_zeta_zero riemannZeta_zero
+
+lemma riemannZeta_def_of_ne_zero {s : ℂ} (hs : s ≠ 0) :
+    riemannZeta s = completedRiemannZeta s / Gammaℝ s := by
+  rw [riemannZeta, hurwitzZetaEven, Function.update_noteq hs, completedHurwitzZetaEven_zero]
+
+/-- The trivial zeroes of the zeta function. -/
+theorem riemannZeta_neg_two_mul_nat_add_one (n : ℕ) : riemannZeta (-2 * (n + 1)) = 0 :=
+  hurwitzZetaEven_neg_two_mul_nat_add_one 0 n
+#align riemann_zeta_neg_two_mul_nat_add_one riemannZeta_neg_two_mul_nat_add_one
+
+/-- Riemann zeta functional equation, formulated for `ζ`: if `1 - s ∉ ℕ`, then we have
+`ζ (1 - s) = 2 ^ (1 - s) * π ^ (-s) * Γ s * sin (π * (1 - s) / 2) * ζ s`. -/
+theorem riemannZeta_one_sub {s : ℂ} (hs : ∀ n : ℕ, s ≠ -n) (hs' : s ≠ 1) :
+    riemannZeta (1 - s) = 2 * (2 * π) ^ (-s) * Gamma s * cos (π * s / 2) * riemannZeta s := by
+  rw [riemannZeta, hurwitzZetaEven_one_sub 0 hs (Or.inr hs'), cosZeta_zero, hurwitzZetaEven_zero]
+#align riemann_zeta_one_sub riemannZeta_one_sub
+
+/-- A formal statement of the **Riemann hypothesis** – constructing a term of this type is worth a
+million dollars. -/
+def RiemannHypothesis : Prop :=
+  ∀ (s : ℂ) (_ : riemannZeta s = 0) (_ : ¬∃ n : ℕ, s = -2 * (n + 1)) (_ : s ≠ 1), s.re = 1 / 2
+#align riemann_hypothesis RiemannHypothesis
+
+/-!
+## Relating the Mellin transform to the Dirichlet series
+-/
+
+theorem completedZeta_eq_tsum_of_one_lt_re {s : ℂ} (hs : 1 < re s) :
+    completedRiemannZeta s =
+      (π : ℂ) ^ (-s / 2) * Gamma (s / 2) * ∑' n : ℕ, 1 / (n : ℂ) ^ s := by
+  have := (hasSum_nat_completedCosZeta 0 hs).tsum_eq.symm
+  simp only [QuotientAddGroup.mk_zero, completedCosZeta_zero] at this
+  simp only [this, Gammaℝ_def, mul_zero, zero_mul, Real.cos_zero, ofReal_one, mul_one, mul_one_div,
+    ← tsum_mul_left]
+  congr 1 with n
+  split_ifs with h <;>
+  simp only [h, Nat.cast_zero, zero_cpow (Complex.ne_zero_of_one_lt_re hs), div_zero, mul_zero]
```

Comment:
```suggestion
  · simp only [h, Nat.cast_zero, zero_cpow (Complex.ne_zero_of_one_lt_re hs), div_zero]
  · rfl
```
I think this is clearer (and worth the additional line). (And `mul_zero`  was apparently not necessary.)